### PR TITLE
Add toggle to disable compiling code that depends on torch-mlir

### PR DESCRIPTION
**Pull request by josel-amd**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,29 +31,35 @@ set(CMAKE_C_STANDARD 11)
...
-      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-      RESULT_VARIABLE STATUS
-      OUTPUT_VARIABLE PACKAGE_DIR
-      ERROR_QUIET)
-    if(NOT STATUS EQUAL "0")
-      message(CHECK_FAIL "not found (install via 'pip install pybind11' or set pybind11_DIR)")
-      return()
+if(NOT DEFINED MLIR_XTEN_WITH_TORCH)
+  set(MLIR_XTEN_WITH_TORCH TRUE)
+endif()
```
Comment (mgehre-amd) [inline CMakeLists.txt]: option(MLIR_XTEN_ENABLE_TORCH "Enable the torch-mlir" ON)

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -7,22 +7,24 @@
...
+llvm_canonicalize_cmake_booleans(MLIR_XTEN_WITH_TORCH)
+
 configure_lit_site_cfg(
   ${CMAKE_CURRENT_SOURCE_DIR}/lit.site.cfg.py.in
   ${CMAKE_CURRENT_BINARY_DIR}/lit.site.cfg.py
   MAIN_CONFIG
   ${CMAKE_CURRENT_SOURCE_DIR}/lit.cfg.py
 )
-
+
```
Comment (mgehre-amd) [inline test/CMakeLists.txt]: nit: trailing space here and below

**Review by mgehre-amd — APPROVED** [review]
Looks good! Did you verify that 
`ninja -C build/third-party/mlir-xten check-aten` works both with and without torch enabled?

File: include/xten/Conversion/CMakeLists.txt [hunk 3]
```diff
@@ -9,6 +9,9 @@
 
 set(LLVM_TARGET_DEFINITIONS Passes.td)
 mlir_tablegen(Passes.h.inc -gen-pass-decls -name XTenConversion)
-add_public_tablegen_target(XTenConversionIncGen)
+
+set(LLVM_TARGET_DEFINITIONS PassesTorch.td)
```
Comment (TinaAMD) [inline include/xten/Conversion/CMakeLists.txt]: Should this be guarded by `MLIR_XTEN_ENABLE_TORCH`?

File: test/CMakeLists.txt (hunk 2 above)
Comment (TinaAMD) [reply test/CMakeLists.txt]: still there

File: include/xten/Transform/PassesTorch.td [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+// (c) Copyright 2022 - 2024 Advanced Micro Devices, Inc. All Rights reserved.
+
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef ATEN_TRANSFORM_PASSES_TORCH
+#define ATEN_TRANSFORM_PASSES_TORCH
+
+include "mlir/Pass/PassBase.td"
+
```
Comment (TinaAMD) [inline include/xten/Transform/PassesTorch.td:18]: Haven't you deleted all these passes (Transforms/ATenDialectOpStats.cpp etc.)? Do we still need the table gen file?

File: include/xten/Transform/PassesTorch.td [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+def ATenOpReport : Pass<"aten-op-report", "ModuleOp"> {
+  let summary = "Generate ATen dialect operation statistics";
+  let constructor = "xilinx::xten::createATenOpReportPass()";
+}
+
+def LowerToLibATen : Pass<"lower-to-libaten", "ModuleOp"> {
+  let summary = "Lower ATen Ops to library calls";
+  let constructor = "xilinx::xten::createLowerToLibATenPass()";
+}
```
Comment (TinaAMD) [inline include/xten/Transform/PassesTorch.td]: Haven't you deleted all these passes (Transforms/ATenDialectOpStats.cpp etc.)? Do we still need the table gen file?

Comment (josel-amd) [reply include/xten/Transform/PassesTorch.td]: True, I think I deleted them after and then I forgot to update here. Will remove.

Comment (josel-amd) [reply include/xten/Transform/PassesTorch.td]: Oh, that file was even not included anywhere.

File: lib/Conversion/CMakeLists.txt [hunk 6]
```diff
@@ -6,19 +6,28 @@
...
+  
+  LINK_COMPONENTS
+  Core
+  
+  LINK_LIBS PUBLIC
 )
+
+if(MLIR_XTEN_ENABLE_TORCH)
+  target_compile_definitions(obj.XTenConversionPasses PRIVATE MLIR_XTEN_ENABLE_TORCH=1)
+endif()
```
Comment (TinaAMD) [inline lib/Conversion/CMakeLists.txt]:
```suggestion
endif()

```

**Review by TinaAMD — APPROVED** [review]
LGTM

**Pull request merged**